src/__experimental__/components/textbox/textbox.d.ts

```diff
@@ -1,87 +1,83 @@
-  readOnly?: boolean;
-  /** Event handler for the change event */
-  onChange?: (ev: React.ChangeEvent<HTMLInputElement>) => void;
-  /** Event handler for the keyDown event */
-  onKeyDown?: (ev: React.KeyboardEvent<HTMLInputElement>) => void;
-  /** Defered callback called after the onChange event */
-  onChangeDeferred?: () => void;
-  /** Integer to determine timeout for defered callback */
-  deferTimeout?: number;
-  /** Label */
+  /** Type of the icon that will be rendered next to the input */
+  inputIcon?: string;
+  /** Optional handler for click event on Textbox icon */
+  iconOnClick?: (ev: React.MouseEvent<HTMLElement>) => void;
+  /** Overrides the default tabindex of the component */
+  iconTabIndex?: string | number;
+  /** The width of the input as a percentage */
+  inputWidth?: number;
+  /** Additional child elements to display before the input */
+  leftChildren?: React.ReactNode;
+  /** Label content */
   label?: string;
-  /** Text applied to label help tooltip */
+  /** A message that the Help component will display */
   labelHelp?: React.ReactNode;
-  /** When true, label is placed in line an input */
+  /** When true label is inline */
   labelInline?: boolean;
   /** Spacing between label and a field for inline label, given number will be multiplied by base spacing unit (8) */
   labelSpacing?: 1 | 2;
-  /** Width of a label in percentage. Works only when labelInline is true */
+  /** Label width */
   labelWidth?: number;
-  /** Width of an input in percentage. Works only when labelInline is true */
-  inputWidth?: number;
-  /** Help content to be displayed under an input */
-  fieldHelp?: React.ReactNode;
-  /** Type of the icon that will be rendered next to the input */
-  children?: React.ReactNode;
-  /** Type of the icon that will be rendered next to the input */
-  inputIcon?: string;
-  /** Additional child elements to display before the input */
-  leftChildren?: React.ReactNode;
-  /** Flag to configure component when in a Form */
-  childOfForm?: boolean;
-  /** Flag to configure component as optional in Form */
-  isOptional?: boolean;
-  /** Indicate that error has occurred
-   * Pass string to display icon, tooltip and red border
-   * Pass true boolean to only display red border
-   */
-  error?: boolean | string;
-  /** Indicate that warning has occurred
-   * Pass string to display icon, tooltip and orange border
-   * Pass true boolean to only display orange border
-   */
-  warning?: boolean | string;
-  /** Indicate additional information
-   * Pass string to display icon, tooltip and blue border
-   * Pass true boolean to only display blue border
-   */
-  info?: boolean | string;
-  /** When true, validation icon will be placed on label instead of being placed on the input */
-  validationOnLabel?: boolean;
-  /** Size of an input */
-  size?: "small" | "medium" | "large";
+  /** Specify a callback triggered on change */
+  onChange?: (ev: React.ChangeEvent<HTMLElement>) => void;
```

Comment:
I think it should be  `ChangeEvent<HTMLInputElement>` instead of `React.ChangeEvent<HTMLElement>`